### PR TITLE
Add & Remove

### PR DESCRIPTION
**Pull request by elmar8287**
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- By default, new tasks should have the property completed set to false and the property index set to the value of the new array length
- All changes to the To-Do List should be saved in local storage.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -25,7 +25,7 @@ <h1>Today's To Do</h1>
     </ul>
 
     <div class="clear-all">
-      <h2>Clear all completed</h2>
+      <h2 id="clear">Clear all completed</h2>
```
Comment (descholar-ceo) [inline src/index.html]: - [x] This button doesn't work, would you fix it?

File: src/index.js [hunk 2]
```diff
@@ -164,3 +155,13 @@ checkboxes.forEach((chbox) => {
...
+
+const clearTasks = task => {
+  task = tasks.completed;
+};
+
+clearAll.addEventListener('click', () => { 
+  tasks.filter(clearTasks);
+  updateLocal(); 
+  addToHTML();
+});
```
Comment (descholar-ceo) [inline src/index.js]: - [x] Kindly fix the clear all button, it's not working

File: src/index.js [hunk 3]
```diff
@@ -59,18 +60,18 @@ const completeTask = (index) => { // function for change the comple status to co
...
-const deleteTask = index => { // function for delete button
+const deleteTask = index => {
   todoItemElems[index].classList.add('deletion');
-  setTimeout(() => { // create a pause for animation when delete
-    tasks.splice(index, 1); // splice delete the index
+  setTimeout(() => { 
+    tasks.splice(index, 1);
     updateLocal();
     addToHTML();
   }, 500);
```
Comment (descholar-ceo) [inline src/index.js:76]: - [x] When we delete a task, all indices should be updated and reordered to follow the normal order, and they should be unique
Screenshot from 2021-08-21 19-01-12

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @elmar8287 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html (hunk 1 above)
Comment (elmar8287) [reply src/index.html]: I fixed it

File: src/index.js (hunk 3 above)
Comment (elmar8287) [reply src/index.js:76]: Fix it

**Review by abruzy — APPROVED** [review]
Hi @elmar8287 👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elmar8287** [comment]
Dear @abruzy! 🥇 

Many thanks for the review. I appriciate your comments 👍 

Many thanks and regards,

Elmar

**Pull request merged**